### PR TITLE
Deprecate SQS queue global setting 

### PR DESCRIPTION
**Pull request by chemamartinez**
## Proposed commit message

Recent tests and reports have revealed that using the same SQS queue for multiple data streams results in data loss even with a very low event load.

For this reason, this pull request updates the affected integrations by forcing users to configure a separate SQS queue for each enabled data stream, as well as updating the related documentation for their knowledge.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Relates https://github.com/elastic/integrations/issues/5526
- Relates https://github.com/elastic/integrations/pull/5687

### CODE REVIEW

File: packages/carbon_black_cloud/_dev/build/docs/README.md [hunk 1]
```diff
@@ -42,17 +42,20 @@ This module has been tested against `Alerts API (v7) [Beta]`, `Alerts API (v6)`,
...
-  - While creating an SQS Queue, please provide the same bucket ARN that has been generated after creating an AWS S3 Bucket.
-3. Set up event notification for an S3 bucket. Follow this [Link](https://docs.aws.amazon.com/AmazonS3/latest/userguide/enable-event-notifications.html).
-  - The user has to perform Step 3 for all the data streams individually, and each time prefix parameter should be set the same as the S3 Bucket List Prefix as created earlier. (for example, `alert_logs/` for the alert data stream.)
-  - For all the event notifications that have been created, select the event type as s3:ObjectCreated:*, select the destination type SQS Queue, and select the queue that has been created in Step 2.
+2. Follow the steps below for each data stream that has been enabled:
+     1. Create an SQS queue
+         - To setup an SQS queue, follow "Step 1: Create an Amazon SQS queue" mentioned in the [Documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/ways-to-add-notification-config-to-bucket.html).
```
Comment (efd6) [inline packages/carbon_black_cloud/_dev/build/docs/README.md:47]:
```suggestion
         - To setup an SQS queue, follow "Step 1: Create an Amazon SQS queue" mentioned in the [Amazon documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/ways-to-add-notification-config-to-bucket.html).
```
(also below)

File: packages/carbon_black_cloud/_dev/build/docs/README.md [hunk 2]
```diff
@@ -42,17 +42,20 @@ This module has been tested against `Alerts API (v7) [Beta]`, `Alerts API (v6)`,
...
+         - While creating an SQS Queue, please provide the same bucket ARN that has been generated after creating an AWS S3 Bucket.
+     2. Setup event notification from the S3 bucket. Follow this [Link](https://docs.aws.amazon.com/AmazonS3/latest/userguide/enable-event-notifications.html). Use the following settings:
```
Comment (efd6) [inline packages/carbon_black_cloud/_dev/build/docs/README.md:49]:
```suggestion
     2. Setup event notification from the S3 bucket using the instructions [here](https://docs.aws.amazon.com/AmazonS3/latest/userguide/enable-event-notifications.html). Use the following settings:
```
(also below)

File: packages/carbon_black_cloud/_dev/build/docs/README.md [hunk 3]
```diff
@@ -42,17 +42,20 @@ This module has been tested against `Alerts API (v7) [Beta]`, `Alerts API (v6)`,
...
+        - Event type: `All object create events` (`s3:ObjectCreated:*`)
+         - Destination: SQS Queue
+         - Prefix (filter): enter the prefix for this data stream, e.g. `alert_logs/`
+         - Select the SQS queue that has been created for this data stream
 
 **Note**:
-  - Credentials for the above AWS S3 and SQS input types should be configured using the [link](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-aws-s3.html#aws-credentials-config).
+  - A separate SQS queue and S3 bucket notification is required for each enabled data stream.
+  - Permissions for the above AWS S3 bucket and SQS queues should be configured as per the [Filebeat S3 input documentation](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-aws-s3.html#_aws_permissions_2)
```
Comment (efd6) [inline packages/carbon_black_cloud/_dev/build/docs/README.md:57]:
```suggestion
  - Permissions for the above AWS S3 bucket and SQS queues should be configured according to the [Filebeat S3 input documentation](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-aws-s3.html#_aws_permissions_2)
```
(also below)

File: packages/carbon_black_cloud/changelog.yml [hunk 4]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "2.4.0"
+  changes:
+    - description: Deprecate global SQS Queue URL to avoid data loss.
+      type: bugfix
+      link: https://github.com/elastic/integrations/pull/10672
```
Comment (efd6) [inline packages/carbon_black_cloud/changelog.yml:6]: Is this breaking? What would happen to a user doing an upgrade? Same question for the other packages here.

Comment (chemamartinez) [reply packages/carbon_black_cloud/changelog.yml:6]: I was also worried about it and I've taken this into account when applying the changes.

That's why I've made no changes to `aws.yml.hbs` files, keeping this block [here](https://github.com/chemamartinez/integrations/blob/deprecate-sqs-queue-global/packages/carbon_black_cloud/data_stream/alert/agent/stream/aws-s3.yml.hbs#L18-L22) even `queue_url` is no longer available in the package's manifest.

With these changes, when a user upgrades the integration, configured queues before the upgrade are kept in the agent policy. The only difference they will experience is that the global queue URL will not be available for new installations.

I made some tests removing that `queue_url` but upgrades got inconsistent, when having a global `queue_url` configured, it got removed after the upgrade and the integration became into an unhealthy state.

Comment (efd6) [reply packages/carbon_black_cloud/changelog.yml:6]: Thanks

**Review by efd6 — APPROVED** [review]

**Pull request merged**